frontend/src/context/Metrics/metricsSlice.tsx

```diff
@@ -78,14 +90,41 @@ export const metricsSlice = createSlice({
       })
     },
 
+    updateMetricsImportedData: (state, action) => {
+      const { crews, cycleTime, doneStatus, classification, deployment, leadTime } = action.payload
+      state.importedData.importedCrews = crews
+      state.importedData.importedCycleTime.importedCycleTimeSettings = cycleTime?.jiraColumns
+      state.importedData.importedCycleTime.importedTreatFlagCardAsBlock = cycleTime?.treatFlagCardAsBlock
+      state.importedData.importedDoneStatus = doneStatus
+      state.importedData.importedClassification = classification
+      state.importedData.importedDeployment = deployment
+      state.importedData.importedLeadTime = leadTime
+    },
+
     updateMetricsState: (state, action) => {
-      const { isProjectCreated, basic } = action.payload
-      state.isProjectCreated = isProjectCreated
-      state.importFile = basic
-      state.users = basic.crews || state.users
-      state.importedCycleTimeSettings = basic.cycleTime?.jiraColumns
-      state.doneColumn = basic.realDone || state.doneColumn
-      state.classification = basic.classification || state.classification
+      const { targetFields, users, jiraColumns, isProjectCreated } = action.payload
+      state.users = isProjectCreated
+        ? users
+        : users?.filter((item: string) => state.importedData.importedCrews.includes(item))
```

Comment:
maybe can use deconstruction method  state.importedData.importedCrews/state.importedData.importedClassification

Reply:
I have fixed it